src/atomate2/common/jobs/neb.py

```diff
@@ -0,0 +1,90 @@
+"""Define tools for analyzing NEB runs."""
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from jobflow import job
+
+if TYPE_CHECKING:
+    from pymatgen.core import Structure
+
+
+class NebInterpolation(Enum):
+    """Methods for interpolating NEB images."""
+
+    LINEAR = "linear"
+    IDPP = "IDPP"
+
+
+@job
+def get_images_from_endpoints(
+    endpoints: tuple[Structure, Structure] | list[Structure],
+    num_images: int,
+    interpolation_method: NebInterpolation = NebInterpolation.LINEAR,
+    **interpolation_kwargs,
+) -> list[Structure]:
+    """
+    Interpolate between two endpoints as a job.
+
+    Parameters
+    ----------
+    endpoints : tuple[Structure,Structure] or list[Structure]
+        A set of two endpoints to interpolate NEB images from.
+    num_images : int
+        The number of images to include in the interpolation.
+    prev_dir : str or Path or None (default)
+        A previous directory to copy outputs from.
+    interpolation_method : .NebInterpolation
+        The method to use to interpolate between images.
+    **interpolation_kwargs
+        kwargs to pass to the interpolation function.
+    """
+    return _get_images_from_endpoints(
+        endpoints,
+        num_images,
+        interpolation_method=NebInterpolation(interpolation_method),
+        **interpolation_kwargs,
+    )
+
+
+def _get_images_from_endpoints(
+    endpoints: tuple[Structure, Structure] | list[Structure],
+    num_images: int,
+    interpolation_method: NebInterpolation = NebInterpolation.LINEAR,
+    **interpolation_kwargs,
+) -> list[Structure]:
+    """
+    Interpolate between two endpoints.
+
+    Parameters
+    ----------
+    endpoints : tuple[Structure,Structure] or list[Structure]
+        A set of two endpoints to interpolate NEB images from.
+    num_images : int
+        The number of images to include in the interpolation.
+    prev_dir : str or Path or None (default)
+        A previous directory to copy outputs from.
+    interpolation_method : .NebInterpolation
+        The method to use to interpolate between images.
+    **interpolation_kwargs
+        kwargs to pass to the interpolation function.
+    """
+    if interpolation_method == NebInterpolation.LINEAR:
+        return endpoints[0].interpolate(
+            endpoints[1], nimages=num_images, **interpolation_kwargs
```

Comment:
I have seen that the pymatgen `interpolate` method behaves in a somewhat weird way: `nimages+1` structures are returned: https://github.com/materialsproject/pymatgen/blob/f9d9fe8e0ce09ef30cc03bcc4e9937d27afd5a6a/src/pymatgen/core/structure.py#L2443
So you get a set of structures which is not `nimages` in total nor `nimages` plus the initial and final structure. Which I found quite unexpected. 
In addition this is at variance with what the `IDPPSolver` does. In that case a total of `nimages+2` structures is returned.
So the meaning of nimages varies depending on the inteprolation method.

I would suggest passing `num_images+1` to `interpolate`. This would maintain consistency among the methods and would also seem more reasonable.